### PR TITLE
Add security section for cache poisoning attack

### PR DESCRIPTION
**Pull request by ierceg**

### CODE REVIEW

**Comment by ianand** [comment]
Task linked: [XDN-5463 Everything that influences rendering should be part of Cache Key](https://app.clickup.com/t/4205457/XDN-5463)

File: guides/cookies.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Cookies
+
+This guide covers the cookies that Moovweb XDN creates and uses.
+
+## Split testing cookies
+
+* `xdn_bucket`: a number between 1 and 100 randomly assigned by Moovweb XDN to all clients connecting to it; it is a long-living non-session cookie allowing predictable cohorts during split testing; its value is not accessible from client-side JavaScript code
```
Comment (markbrocato) [inline guides/cookies.md:7]:
```suggestion
* `xdn_bucket`: a number between 1 and 100 randomly assigned by the Moovweb XDN to all clients connecting to it. It is a long-lived non-session cookie allowing predictable cohorts during split testing. Its value is not accessible from client-side JavaScript code.
```

File: guides/cookies.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+* `xdn_destination`: the name of the user experience to which the split testing has assigned the user based on the value of `xdn_bucket`; its value is accessible to client-side JavaScript code and can thus be safely levaraged from client-side metrics
```
Comment (markbrocato) [inline guides/cookies.md:8]:
```suggestion
* `xdn_destination`: the name of the experience to which split testing has assigned the user based on the value of `xdn_bucket`. Its value is accessible to client-side JavaScript code and can thus be safely included in client-side metrics.
```

File: guides/caching.md [hunk 3]
```diff
@@ -64,6 +64,8 @@ When [POST and other non-GET/HEAD](#section_caching_responses_for_post_and_other
 - Request HTTP method
 - Request body
 
+To ensure that you are resilient to [cache poisoning attacks](security#section_cache_poisoning) you must ensure that every request header that influences the rendering of the content is also part of the custom cache key.
```
Comment (markbrocato) [inline guides/caching.md:67]:
```suggestion
To ensure that your site is resilient to [cache poisoning attacks](security#section_cache_poisoning) every request header that influences the rendering of the content must be included in your custom cache key. 
```

File: guides/security.md [hunk 4]
```diff
@@ -55,3 +55,24 @@ store them in environment variables, then access them in your code from `process
...
 
 ![networking](/images/security/environment-variables.png)
+
+## Cache poisoning
+
+[Cache poisoning attack](https://owasp.org/www-community/attacks/Cache_Poisoning) is described by OWASP as:
+
+> The impact of a maliciously constructed response can be magnified if it is cached either by a web cache used by multiple users or even the browser cache of a single user. If a response is cached in a shared web cache, such as those commonly found in proxy servers, then all users of that cache will continue to receive the malicious content until the cache entry is purged.
+
+To guard against this attack you must ensure that all the request parameters that influence the rendering of the content are part of your [custom cache key](caching#section_customizing_the_cache_key). XDN will [automatically include](caching#section_cache_key) `host` header and URL but including other request headers and cookies are your responsibility.
```
Comment (markbrocato) [inline guides/security.md:65]:
```suggestion
To guard against this attack you must ensure that all the request parameters that influence the rendering of the content are part of your [custom cache key](caching#section_customizing_the_cache_key). The XDN will [automatically include](caching#section_cache_key) the `host` header and URL. Including other request headers and cookies are your responsibility.
```

**Review by markbrocato — APPROVED** [review]
See my suggestions

**Pull request merged**